[CI] flaky test failure - o.o.indices.stats.IndexStatsIT.testFilterCacheStats

Caught from PR https://github.com/opensearch-project/OpenSearch/pull/2424#issuecomment-1071889507:

❌   Gradle Check failure https://github.com/opensearch-project/OpenSearch/commit/90f143afca0b28eb9d15d3b210bcc8339816230a
[Log 3504](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_3504.log)

[Reports 3504](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_3504_reports.zip)

In log 3504:
```
> Task :server:internalClusterTest

REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.indices.stats.IndexStatsIT.testFilterCacheStats" -Dtests.seed=D48688166EC0CB86 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=fr -Dtests.timezone=GMT -Druntime.java=17

org.opensearch.indices.stats.IndexStatsIT > testFilterCacheStats FAILED
    java.lang.AssertionError: expected:<0> but was:<1>
        at __randomizedtesting.SeedInfo.seed([D48688166EC0CB86:B1B97553DEC8CCFF]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.indices.stats.IndexStatsIT.testFilterCacheStats(IndexStatsIT.java:1225)
```

--- a/server/src/main/java/org/opensearch/index/engine/InternalEngine.java
+++ b/server/src/main/java/org/opensearch/index/engine/InternalEngine.java
@@ -1289,6 +1289,7 @@ public class InternalEngine extends Engine {
             }
         } else {
             try (Searcher searcher = acquireSearcher("assert doc doesn't exist", SearcherScope.INTERNAL)) {
+                searcher.setQueryCache(null);
                 final long docsWithId = searcher.count(new TermQuery(index.uid()));
                 if (docsWithId > 0) {
                     throw new AssertionError("doc [" + index.id() + "] exists [" + docsWithId + "] times in index");

